atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/CqlClientImpl.java

```diff
@@ -35,7 +35,8 @@
 import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
 import java.util.concurrent.Executor;
 
-public final class CqlClientImpl implements CqlClient {
+@SuppressWarnings({"FinalClass", "Not final for spying in tests"})
```

Comment:
Reasonable! I copied this from the top of `CassandraKeyValueServiceImpl` 🗡️ 
Avoiding spying is a bit of a pain, since I'm trying to artificially "create" a race condition in that the cql client seems valid even though the session is closed, and therefore need to force the CQL Client into believing it's valid without breaking the Session instance. Would definitely appreciate an alternative if there is one, probably overlooking somehting